client/my-sites/checkout/checkout/index.jsx

```diff
@@ -447,9 +449,10 @@ export class Checkout extends React.Component {
 			if ( this.props.redirectToPageBuilder ) {
 				return getEditHomeUrl( selectedSiteSlug );
 			}
-			const destination = abtest( 'improvedOnboarding' ) === 'main' ? 'checklist' : 'view';
 
-			return `/${ destination }/${ selectedSiteSlug }${ queryParam }`;
+			const destination = retrieveSignupDestination();
+
+			return `${ destination }${ queryParam }`;
```

Comment:
Do we need a fallback for the (unlikely) case that the cookie value isn't saved?

Reply:
Fallback added in [cbf8475](https://github.com/Automattic/wp-calypso/pull/34156/commits/cbf8475fec453f197d97d35f2b5f3bc5bb3943f4).

> Also wondering if it's too defensive to remove the cookie somewhere after we've read it or before we set it? Unless we need it again for future use.

I was oscillating on this point too, if I should clear it or not. I couldn't see a harm in keeping the cookie value, but maybe it's just safer to throw that value away when we don't need it? I'll bite it, and add that fix :)

Reply:
Fixed in [0f2db23](https://github.com/Automattic/wp-calypso/pull/34156/commits/0f2db23172305e129fdf2348b4526890b67b2144).

Reply:
> maybe it's just safer to throw that value away when we don't need it? I'll bite it, and add that fix :)

Sorry, I was just musing out loud. Publicly. 😄 

Your judgement on this is way more sophisticated than mine, and I didn't want to give them impression that it was a blocker. 

But since, it doesn't hurt to delete it, bite away 🍭